test/e2e/tests/test_transformjob.py

```diff
@@ -23,142 +23,178 @@
 
 RESOURCE_PLURAL = "transformjobs"
 
 
-@pytest.fixture(scope="module")
-def xgboost_transformjob(sagemaker_client):
-    # Create model using boto3 for TransformJob
-    transform_model_file = (
-        f"s3://{get_bootstrap_resources().DataBucketName}/sagemaker/batch/model.tar.gz"
-    )
-    model_name = random_suffix_name("xgboost-model", 32)
-
-    create_response = sagemaker_client.create_model(
-        ModelName=model_name,
-        PrimaryContainer={
-            "Image": REPLACEMENT_VALUES["XGBOOST_IMAGE_URI"],
-            "ModelDataUrl": transform_model_file,
-            "Environment": {},
-        },
-        ExecutionRoleArn=REPLACEMENT_VALUES["SAGEMAKER_EXECUTION_ROLE_ARN"],
+@pytest.fixture(scope="function")
+def generate_job_names():
+    transform_resource_name = random_suffix_name("xgboost-transformjob", 27)
+    model_resource_name = "model" + transform_resource_name
+
+    yield (transform_resource_name, model_resource_name)
+
+
+@pytest.fixture(scope="function")
+def xgboost_model_for_transform(generate_job_names):
+    (transform_resource_name, model_resource_name) = generate_job_names
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["MODEL_NAME"] = model_resource_name
+
+    reference, _, resource = create_sagemaker_resource(
+        resource_plural=cfg.MODEL_RESOURCE_PLURAL,
+        resource_name=model_resource_name,
+        spec_file="xgboost_model",
+        replacements=replacements,
     )
-    logging.debug(create_response)
+    assert resource is not None
+    assert k8s.get_resource_arn(resource) is not None
+
+    yield (transform_resource_name, model_resource_name)
 
-    # Check if the model is created successfully
-    describe_model_response = sagemaker_client.describe_model(ModelName=model_name)
-    assert describe_model_response["ModelName"] is not None
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted
 
-    resource_name = random_suffix_name("xgboost-transformjob", 32)
 
-    # Use the model created above
+# TODO: This method can also move to a common file.
+@pytest.fixture(scope="function")
+def xgboost_transformjob(xgboost_model_for_transform):
+    (transform_resource_name, model_resource_name) = xgboost_model_for_transform
     replacements = REPLACEMENT_VALUES.copy()
-    replacements["MODEL_NAME"] = model_name
-    replacements["TRANSFORM_JOB_NAME"] = resource_name
+    replacements["MODEL_NAME"] = model_resource_name
+    replacements["TRANSFORM_JOB_NAME"] = transform_resource_name
 
-    reference, spec, resource = create_sagemaker_resource(
+    reference, _, resource = create_sagemaker_resource(
         resource_plural=RESOURCE_PLURAL,
-        resource_name=resource_name,
+        resource_name=transform_resource_name,
         spec_file="xgboost_transformjob",
         replacements=replacements,
     )
+
     assert resource is not None
+    assert k8s.get_resource_arn(resource) is not None
```

Comment:
I think this check should be here because it means we are expecting sdkCreate to have succeeded. if it fails for some reason, the assert would fail and resource won't get cleaned up 
so, we should have a try-catch block in the fixture as well or better move this check to the test class